test/extended/images/append.go

```diff
@@ -0,0 +1,134 @@
+package images
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/origin/pkg/image/dockerlayer"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+func cliPodWithPullSecret(cli *exutil.CLI, shell string) *kapiv1.Pod {
+	err := exutil.WaitForServiceAccount(cli.KubeClient().Core().ServiceAccounts(cli.Namespace()), "builder")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	sa, err := cli.KubeClient().Core().ServiceAccounts(cli.Namespace()).Get("builder", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(sa.ImagePullSecrets).NotTo(o.BeEmpty())
+	pullSecretName := sa.ImagePullSecrets[0].Name
+
+	// best effort to get the format string for the release
+	router, err := cli.AdminAppsClient().Apps().DeploymentConfigs("default").Get("router", metav1.GetOptions{})
+	if err != nil {
+		g.Skip(fmt.Sprintf("Unable to find router in order to query format string: %v", err))
```

Comment:
this seems fragile and likely to lead to this test being skipped w/o anyone noticing.